### PR TITLE
9577 replace roles allowed

### LINKED ISSUE #9577: Replace @RolesAllowed by own @RightsAllowed annotation
### Problem Description

Proven performance problem (#9559) when checking `javax.annotation.security.@RolesAllowed`

### Proposed Change

Replace `@RolesAllowed` by own `@RightsAllowed` annotation and check with interceptor.

### Acceptance Criteria

- [x] `@PermitAll` and `@DenyAll` is also considered.
- [x] UserRight._SYSTEM is also considered (e.g. completeness calculation of case is working)
- [x] Add archUnit test to make sure `@RolesAllowed` is not used in backend (in sormas-rest it's fine)

### PR DESCRIPTION
**Pull request by MartinWahnschaffe**
Fixes #9577

### CODE REVIEW

File: sormas-backend/src/test/java/de/symeda/sormas/backend/ArchitectureTest.java [hunk 1]
```diff
@@ -291,6 +292,14 @@ private void assertFacadeEjbAnnotated(Class<?> facadeEjbClass, AuthMode authMode
 
 			methodChecks.check(classes);
 		}
+
+		// don't use RolesAllowed anymore, until performance problem is fixed
```
Comment (StefanKock) [inline sormas-backend/src/test/java/de/symeda/sormas/backend/ArchitectureTest.java:296]: This code block only checks EJBs that are given by a test. I would rather check all classes in the module annotated with `@Singleton`/`@Stateless` that there is no usage of `@RolesAllowed`.

**Review by StefanKock — CHANGES_REQUESTED** [review]
Looks good to me so far, but the check for `@RolesAllowed` should be more general in my opinion.

File: sormas-backend/src/test/java/de/symeda/sormas/backend/ArchitectureTest.java (hunk 1 above)
Comment (MartinWahnschaffe) [reply sormas-backend/src/test/java/de/symeda/sormas/backend/ArchitectureTest.java]: Guess there is no harm in simply not allowing it for all classes and methods in the package.

**Review by StefanKock — APPROVED** [review]
Checking for not using `@RolesAllowed` looks way better and easier than I expected.

**Pull request merged**